web_widget_text_markdown/controllers/web_main.py

```diff
@@ -0,0 +1,44 @@
+import base64
+import unicodedata
+import logging
+from odoo import http, _
+from odoo.addons.web.controllers.main import serialize_exception
+from odoo.http import request
+_logger = logging.getLogger(__name__)
+
+
+class Home(http.Controller):
+    # This is a cloned method from web>controllers>main 'upload_attachment'
+    # that only change the output
+    @http.route(
+        '/web_widget_text_markdown/upload_attachment',
+        type='http', auth="user")
+    @serialize_exception
+    def upload_markdown_attachment(self, callback, model, id, ufile):
+        files = request.httprequest.files.getlist('ufile')
+        Model = request.env['ir.attachment']
+        args = []
+        for ufile in files:
+
+            filename = ufile.filename
+            if request.httprequest.user_agent.browser == 'safari':
+                # Safari sends NFD UTF-8 (where é is composed by 'e' and
+                # [accent]) we need to send it the same stuff, otherwise it'll
+                # fail
+                filename = unicodedata.normalize('NFD', ufile.filename)
+
+            try:
+                attachment = Model.create({
+                    'name': filename,
+                    'datas': base64.encodestring(ufile.read()),
+                    'datas_fname': filename,
+                    'res_model': model,
+                    'res_id': int(id)
```

Comment:
pls rename `id` to `res_id`.
Also, I wonder: when you are creating a record you won't have a real id here... hence you'll lose the reference.